### PR TITLE
[Typescript Pattern 2]Test a function handler locally that calls cloud services

### PR DESCRIPTION
**Pull request by chankina**
Issue #135

Github Issue link: https://github.com/aws-samples/serverless-test-samples/issues/135

Test a function handler locally that calls cloud services

*Description of changes:* Adding a new typescript pattern

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: typescript-test-samples/lambda-handler-dynamodb/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+---
+
+## About this Example
+
+This example contains an [AWS Lambda](https://aws.amazon.com/lambda/) function and [Amazon DynamoDB](https://aws.amazon.com/dynamodb/) table as core resources.
+
+AWS CDK is used to deploy the AWS Lambda function and Amazon DynamoDB table. The AWS Lambda function generates an item in the JSON format and writes this item to the DynamoDB table using the AWS SDK. The DynamoDB table item is a JSON object with format:
+
+```json
+{
```
Comment (brnkrygs) [inline typescript-test-samples/lambda-handler-dynamodb/README.md:61]: This doesn't seem to be the right item structure. It should match what the handler is writing in `app.ts` on line 22+.

File: typescript-test-samples/lambda-handler-dynamodb/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+### Run the Integration Tests
+
+[integ-handler.test.ts](test/integration/integ-handler.test.ts) 
+
+Deploy the full stack:
+
+```shell
+lambda-handler-dynamodb$ npm install
+lambda-handler-dynamodb$ cdk deploy
```
Comment (brnkrygs) [inline typescript-test-samples/lambda-handler-dynamodb/README.md:90]: I get an error trying to run this. Looks like the `/lib` and/or `/bin` directory is missing from the commit. Its likely it was ignored by the `.gitignore`. You may have to explicitly tell git to add it.

For example:
`git add lib/lambda-handler-dynamic-stack.ts`

File: typescript-test-samples/lambda-handler-dynamodb/test/integration/integ-handler.test.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+/**
```
Comment (brnkrygs) [inline typescript-test-samples/lambda-handler-dynamodb/test/integration/integ-handler.test.ts:1]: Add copyright lines at the top of this file (see `/src/app.ts` for example)

File: typescript-test-samples/lambda-handler-dynamodb/package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  "version": "0.1.0",
+  "bin": {
+    "lambda-handler-dynamodb": "bin/lambda-handler-dynamodb.js"
+  },
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc -w",
+    "test": "jest",
+    "cdk": "cdk",
+    "test:unit": "jest --selectProjects unit",
```
Comment (brnkrygs) [inline typescript-test-samples/lambda-handler-dynamodb/package.json:12]: `test:unit` and `test:integration` are not configured for this project, we should drop them.

File: typescript-test-samples/lambda-handler-dynamodb/test/integration/integ-handler.test.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    DynamoDBDocumentClient,
+    DeleteCommand,
+    GetCommandInput,
+    GetCommand
+} from "@aws-sdk/lib-dynamodb";
+import { eventJSON } from "../../src/events/event-data";
+import { lambdaHandler } from "../../src/app";
+
+
+//process.env.DatabaseTable = "LambdaHandlerDynamodbStack-DynamoTableB2B22E15-HCH4NYL0QMJZ";
```
Comment (brnkrygs) [inline typescript-test-samples/lambda-handler-dynamodb/test/integration/integ-handler.test.ts:18]: Drop this, not needed.

**Review by brnkrygs — CHANGES_REQUESTED** [review]
Thanks for your effort on this @chankina!

Need a couple of tweaks to config files, and need the CDK app itself added to the repository.

Finally, we also need a `metadata.json` file alongside the `README.md` file, so that we can publish the sample on Serverlessland. Check out this file as an example:

https://github.com/aws-samples/serverless-test-samples/blob/main/typescript-test-samples/kinesis-lambda-dynamodb/metadata.json

File: typescript-test-samples/lambda-handler-dynamodb/README.md (hunk 1 above)
Comment (chankina) [reply typescript-test-samples/lambda-handler-dynamodb/README.md:61]: Updated the json structure in ReadMe. Please review.

File: typescript-test-samples/lambda-handler-dynamodb/README.md (hunk 2 above)
Comment (chankina) [reply typescript-test-samples/lambda-handler-dynamodb/README.md:91]: Yeah the gitignore in the main repo ignored lib/ file so manually added in this pattern. You should be able to run the cdk app now.

**Comment by chankina** [comment]
Thanks for reviewing it. I updated based on all your comments. Please review. Thanks!

File: typescript-test-samples/lambda-handler-dynamodb/README.md (hunk 2 above)
Comment (brnkrygs) [reply typescript-test-samples/lambda-handler-dynamodb/README.md:91]: Still getting an error here unfortunately. Might need the `*.ts* file from under `/bin` added to the repo too.

File: typescript-test-samples/lambda-handler-dynamodb/lib/lambda-handler-dynamodb-stack.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import * as cdk from 'aws-cdk-lib';
```
Comment (brnkrygs) [inline typescript-test-samples/lambda-handler-dynamodb/lib/lambda-handler-dynamodb-stack.ts:1]: Add copyright lines at the top of this file (see /src/app.ts for example)

Sorry, missed this one on the last review.

**Review by brnkrygs — CHANGES_REQUESTED** [review]
Still getting a CDK error. Check out comment thread on README for details.

**Review by brnkrygs — APPROVED** [review]
Comments addressed, app builds and deploys and tests pass. Approved.

**Pull request merged**